runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsOverlay.java

```diff
@@ -53,20 +56,57 @@
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		LocalPoint dest = client.getLocalDestinationLocation();
+		if (config.highlightCurrentLocation())
+		{
+			LocalPoint position = LocalPoint.fromWorld(client, client.getLocalPlayer().getWorldLocation());
+			highlightTile(graphics, position, config.highlightCurrentLocationColor());
+		}
+
+		if (config.highlightDestination())
+		{
+			highlightTile(graphics, client.getLocalDestinationLocation(), config.highlightDestinationColor());
+		}
+
+		if (config.highlightMousePosition())
+		{
+			Tile[][][] tiles = client.getRegion().getTiles();
```

Comment:
Agreed however when I dug through the client code I was unable to find anything that suggested they were converting mouse canvas position to in-game tile. Rather it seems like they do something similar and convert tile to canvas area (like what Perspective.java is able to do) and checking bounds of the mouse position/clicks.

What I found when looking at what happens when a menu is drawn on a clicked tile was:
1. When building menu, "Walk Here" entry added at [BuildType#76](../tree/master/runescape-client/src/main/java/BuildType.java#L76)
2. The call to VertexNormal eventually leads to [class98#189](../tree/master/runescape-client/src/main/java/class98.java#L189), where the menu is stored in the client options
    * var0 ="Walk Here"
    * var1 = ""
    * var2 = the action (23 for Walk here)
    * var3 = 0
    * var4 is the X coordinate of the mouse click
    * var5 the Y coordinate of the mouse click
3. Now when the menu action is performed, we are led to [FileOnDisk#526](../tree/master/runescape-client/src/main/java/FileOnDisk.java#L526)
4. Region.method2938 is here: [Region#1037](../tree/master/runescape-client/src/main/java/Region.java#L1037)
    * var2 and var3 are the X and Y coordinates

So when I look for what writes to the selectedRegionTileX/Y to perform the translation, I'm led to methods like [Region.drawTileOverlay](../tree/master/runescape-client/src/main/java/Region.java#L1951) which is called from draw(Tile, ...) and performs something similar to what I do (I think, obfuscated) in method2947 [here](../tree/master/runescape-client/src/main/java/Region.java#L2560).

--

I'm not all that comfortable yet with how the canvas layer coordinates interact with camera and in-game coordinates yet to figure out how to translate the mouse co-ordinates to the tile since there is nothing in Perspective.java to help solve this issue.

An alternative optimization I could make would be to save the current tile the mouse is over and only do the recalculation whenever the mouse coordinates change.